### ISSUE TITLE
RowNumberForPaging, two tables and columns with same name

### ISSUE THREAD

**Issue by sepplK** [aspnet/EntityFramework#5641]
### Steps to reproduce
```
optionsBuilder.UseSqlServer("connectionString", o=>
{
    o.UseRowNumberForPaging();
});

var orders = (from ba in dc.BestellungenArtikel
                join b in dc.Bestellungen on ba.Bestellung equals b.Id
                orderby b.Eingang descending
                select new
                {
                    OrderId = b.Id,
                    ArtId = ba.Id,
                    ArtName = ba.Name
                }).Skip(10).Take(20).ToList();
```

```
IDs equal:
{ OrderId = 387945, ArtId = 387945, ArtName = Guten Morgen } 
{ OrderId = 387944, ArtId = 387944, ArtName = Romantisches Duett }
```

```
exec sp_executesql N'SELECT [t].[ID], [t].[ID] AS [c0], [t].[name]
FROM (
    SELECT [b].[ID], [ba].[ID] AS [c0], [ba].[name], ROW_NUMBER() OVER(ORDER BY [b].[eingang] DESC) AS [__RowNumber__]
    FROM [bestellungen_artikel] AS [ba]
    INNER JOIN [bestellungen] AS [b] ON [ba].[bestellung] = [b].[ID]
) AS [t]
WHERE ([t].[__RowNumber__] > @__p_0) AND ([t].[__RowNumber__] <= (@__p_0 + @__p_1))',N'@__p_0 int,@__p_1 int',@__p_0=10,@__p_1=20
```

Without Paging:

```
var orders = (from ba in dc.BestellungenArtikel
                join b in dc.Bestellungen on ba.Bestellung equals b.Id
                orderby b.Eingang descending
                where b.Id == 387945 || b.Id == 387944
                select new
                {
                    OrderId = b.Id,
                    ArtId = ba.Id,
                    ArtName = ba.Name
                }).ToList();
```

```
IDs OK:
{ OrderId = 387945, ArtId = 758860, ArtName = Guten Morgen }
{ OrderId = 387944, ArtId = 758859, ArtName = Romantisches Duett }
```

```
SELECT [b].[ID], [ba].[ID], [ba].[name]
FROM [bestellungen_artikel] AS [ba]
INNER JOIN [bestellungen] AS [b] ON [ba].[bestellung] = [b].[ID]
WHERE [b].[ID] IN (387945, 387944)
ORDER BY [b].[eingang] DESC
```

### The issue
OrderId and ArtId is the same after using RowNumberPaging.
Select in paged query: [t].[ID], [t].[ID] AS [c0], [t].[name] VS [b].[ID], [ba].[ID], [ba].[name]

### Further technical details

EF Core version: RC2
SQL version: 2008 R2

**Comment by maumar** [comment]
Most likely duplicate of https://github.com/aspnet/EntityFramework/issues/4739

**Comment by sepplK** [comment]
workarround for RC2:

https://github.com/sepplK/EntityFramework-1.0.0-rc2-pi/commit/fb363f755a9da9d16a2a4112aa17509078ef32f3

**Comment by divega** [comment]
Not sure if it is the same as #4739. That one produces a `NullReferenceException`. This one returns wrong data from the query.

**Comment by divega** [comment]
@maumar explained to me that although the symptoms are different it looks like another manifestation the same underlying issue as #4739.

**Comment by smitpatel** [comment]
Both have same cause. The projections are incorrectly renamed while translating for row number paging. In this case it doesn't know exception because the property which gets wrong value is same type is the other property . So incorrect result. It throws NRE in #4739 because null value is being assigned to non-null property.

**Comment by maumar** [comment]
fixed in 11c18e6f03997c381ff16def574fd08f4d352138

**Comment by ajcvickers** [comment]
@sepplK If you are still using RowNumberForPaging, could you please comment on issue #13959 and let us know why?
